src/main/java/com/labsynch/labseer/service/StandardizationServiceImpl.java

```diff
@@ -0,0 +1,524 @@
+	}
+
+
+	@Override
+	public int dupeCheckStandardizationStructures() throws CmpdRegMolFormatException{
+		List<Long> qcIds = StandardizationDryRunCompound.findAllIds().getResultList();
+		int totalCount = qcIds.size();
+		logger.debug("number of compounds found in dry run table: " + totalCount);
+		int totalNewDuplicateCount = 0;
+		int totalExistingDuplicateCount = 0;
+		if (qcIds.size() > 0){
+			int[] hits;	
+			StandardizationDryRunCompound qcCompound;
+			String newDuplicateCorpNames = "";
+			String oldDuplicateCorpNames = "";
+			int newDupeCount = 0;
+			int oldDuplicateCount = 0;
+			
+			float percent = 0;
+			int p = 1;
+		    float previousPercent = percent;
+			previousPercent = percent;
+			for (Long qcId : qcIds){
+				boolean firstNewDuplicateHit = true;
+				boolean firstOldDuplicateHit = true;
+				qcCompound = StandardizationDryRunCompound.findStandardizationDryRunCompound(qcId);
+				logger.debug("query compound: " + qcCompound.getCorpName());
+				if(qcCompound.getNewMolWeight() == 0) {
+					logger.debug("mol has a weight of 0 - skipping");			
+				} else {
+					hits = chemStructureService.searchMolStructures(qcCompound.getMolStructure(), StructureType.DRY_RUN, SearchType.DUPLICATE_TAUTOMER);
+					newDupeCount = hits.length;
+					qcCompound.setChangedStructure(true);
+					for (int hit:hits){
+						List<StandardizationDryRunCompound> searchResults = StandardizationDryRunCompound.findStandardizationDryRunCompoundsByCdId(hit).getResultList();
+						for (StandardizationDryRunCompound searchResult : searchResults){
+							if (searchResult.getCorpName().equalsIgnoreCase(qcCompound.getCorpName())){
+								qcCompound.setChangedStructure(false);
+								newDupeCount = newDupeCount-1;
+							} else {
+								if (StringUtils.equals(searchResult.getStereoCategory(), qcCompound.getStereoCategory())
+										&& StringUtils.equalsIgnoreCase(searchResult.getStereoComment(), qcCompound.getStereoComment())){
+									if (!firstNewDuplicateHit) newDuplicateCorpNames = newDuplicateCorpNames.concat(";");
+									newDuplicateCorpNames = newDuplicateCorpNames.concat(searchResult.getCorpName());
+									firstNewDuplicateHit = false;
+									logger.debug("found new dupe parents");
+									logger.debug("query: " + qcCompound.getCorpName() + "     dupe: " + searchResult.getCorpName());
+									totalNewDuplicateCount++;
+								} else {
+									logger.debug("found different stereo codes and comments");
+								}
+							}
+						}
+					}
+					
+					hits = chemStructureService.searchMolStructures(qcCompound.getMolStructure(), StructureType.PARENT, SearchType.DUPLICATE_TAUTOMER);
+					oldDuplicateCount = hits.length;
+					for (int hit:hits){
+						List<Parent> searchResults = Parent.findParentsByCdId(hit).getResultList();
+						for (Parent searchResult : searchResults){
+							if (searchResult.getCorpName().equalsIgnoreCase(qcCompound.getCorpName())){
+								oldDuplicateCount = oldDuplicateCount-1;
```

Comment:
I think the "changedStructure" logic should be here instead.

Reply:
Yes, thanks. Moved it here instead.

Reply:
I just caused a lot of diffs but to fix this i ran a format using my vs code formatter.  There were lots of indentation problems with this file.